### PR TITLE
ci: build examples and deply in a subfolder

### PR DESCRIPTION
**Pull request by jakubgs**
Addition of building of examples site and publishing under `/examples/`.

* https://js-waku.wakuconnect.dev/examples/web-chat/
* https://js-waku.wakuconnect.dev/examples/eth-pm/
* https://js-waku.wakuconnect.dev/examples/eth-pm-wallet-encryption/

### CODE REVIEW

**Comment by jakubgs** [comment]
I'm getting some interesting errors:
```
+ yarn build

yarn run v1.22.17
$ run-s build:**
$ tsc -p tsconfig.json
$ tsc -p tsconfig.dev.json
src/lib/discovery/enrtree.spec.ts:20:14 - error TS2571: Object is of type 'unknown'.
20       expect(e.toString()).includes(
                ~

src/lib/discovery/enrtree.spec.ts:30:14 - error TS2571: Object is of type 'unknown'.
30       expect(e.toString()).includes('Unable to verify ENRTree root signature');
                ~

src/lib/discovery/enrtree.spec.ts:38:14 - error TS2571: Object is of type 'unknown'.
38       expect(e.toString()).includes('Could not parse ENRTree root entry');
                ~

src/lib/discovery/enrtree.spec.ts:54:14 - error TS2571: Object is of type 'unknown'.
54       expect(e.toString()).includes(
                ~
...
```
https://ci.status.im/blue/organizations/jenkins/website%2Fjs-waku.wakuconnect.dev/detail/js-waku.wakuconnect.dev/26/pipeline/

**Comment by D4nte** [comment]
> I'm getting some interesting errors:
> 
> ```
> + yarn build
> 
> yarn run v1.22.17
> $ run-s build:**
> $ tsc -p tsconfig.json
> $ tsc -p tsconfig.dev.json
> src/lib/discovery/enrtree.spec.ts:20:14 - error TS2571: Object is of type 'unknown'.
> 20       expect(e.toString()).includes(
>                 ~
> 
> src/lib/discovery/enrtree.spec.ts:30:14 - error TS2571: Object is of type 'unknown'.
> 30       expect(e.toString()).includes('Unable to verify ENRTree root signature');
>                 ~
> 
> src/lib/discovery/enrtree.spec.ts:38:14 - error TS2571: Object is of type 'unknown'.
> 38       expect(e.toString()).includes('Could not parse ENRTree root entry');
>                 ~
> 
> src/lib/discovery/enrtree.spec.ts:54:14 - error TS2571: Object is of type 'unknown'.
> 54       expect(e.toString()).includes(
>                 ~
> ...
> ```
> 
> [ci.status.im/blue/organizations/jenkins/website%2Fjs-waku.wakuconnect.dev/detail/js-waku.wakuconnect.dev/26/pipeline](https://ci.status.im/blue/organizations/jenkins/website%2Fjs-waku.wakuconnect.dev/detail/js-waku.wakuconnect.dev/26/pipeline/)

Please use `npm`. The errors are related to the  latest TypeScript version. We do not use it in the repo (yet).
By using yarn it installed the latest version of TypeScript, with `npm`, it should use the correct version.

File: ci/Jenkinsfile.gh-pages [hunk 1]
```diff
@@ -28,6 +28,16 @@ pipeline {
       }
     }
 
+    stage('Examples') {
```
Comment (D4nte) [inline ci/Jenkinsfile.gh-pages:31]: Not sure if you have done it but you first need to run, once, from root:

```
npm ci
npm run build
```

And then go in each example folders to run (can be done in parallel)

```
npm ci
npm run build
````

**Comment by jakubgs** [comment]
>The errors are related to the latest TypeScript version. We do not use it in the repo (yet).
>By using yarn it installed the latest version of TypeScript, with npm, it should use the correct version.

What? Why would it install the wrong version if we explicitly specify the verison in `package.json`?
https://github.com/status-im/js-waku/blob/671c757fa21c06de37c7ad6e2d4aa877f49a688b/examples/eth-pm/package.json#L23
We can use `npm` but this is some weird shit.

File: ci/Jenkinsfile.gh-pages (hunk 1 above)
Comment (jakubgs) [reply ci/Jenkinsfile.gh-pages:31]: That's what I'm doing, just with yarn, look at the function.

**Comment by D4nte** [comment]
> >The errors are related to the latest TypeScript version. We do not use it in the repo (yet).
> >By using yarn it installed the latest version of TypeScript, with npm, it should use the correct version.
> 
> What? Why would it install the wrong version if we explicitly specify the verison in `package.json`?
> https://github.com/status-im/js-waku/blob/671c757fa21c06de37c7ad6e2d4aa877f49a688b/examples/eth-pm/package.json#L23
> We can use `npm` but this is some weird shit.

Good point. I'll investigate more.

**Comment by jakubgs** [comment]
Weird failure for `eth-pm`:
```

+ npm ci
npm ERR! code ENOTEMPTY
npm ERR! syscall rename
npm ERR! path /home/jenkins/workspace/website/js-waku.wakuconnect.dev/node_modules/sane
npm ERR! dest /home/jenkins/workspace/website/js-waku.wakuconnect.dev/node_modules/.sane-dx8KQgp8
npm ERR! errno -39
npm ERR! ENOTEMPTY: directory not empty, rename '/home/jenkins/workspace/website/js-waku.wakuconnect.dev/node_modules/sane' -> '/home/jenkins/workspace/website/js-waku.wakuconnect.dev/node_modules/.sane-dx8KQgp8'
npm ERR! A complete log of this run can be found in:
npm ERR!     /home/jenkins/.npm/_logs/2022-01-25T11_19_04_344Z-debug.log
script returned exit code 217
```
Wat?

File: ci/Jenkinsfile.gh-pages (hunk 1 above)
Comment (D4nte) [reply ci/Jenkinsfile.gh-pages:32]: I think you are missing a `npm run build` at the root. If I understand correctly, you are only running `yarn run doc` at the root.

I tried to test myself: https://github.com/status-im/js-waku/pull/437 but I could not find how to run a jenkins with my changes.

File: ci/Jenkinsfile.gh-pages [hunk 2]
```diff
@@ -28,13 +29,47 @@ pipeline {
       }
     }
 
-    stage('Publish Prod') {
-      when { expression { GIT_BRANCH.endsWith('docs-build') } }
+    stage('Examples') {
+      steps {
+        script {
+          parallel genExampleBuilds([
+            'js-waku', 'web-chat', 'eth-pm', 'eth-pm-wallet'
```
Comment (D4nte) [inline ci/Jenkinsfile.gh-pages:36]:
```suggestion
            'web-chat', 'eth-pm', 'eth-pm-wallet-encryption'
```

**Review by D4nte — APPROVED** [review]
LGTM

File: ci/Jenkinsfile.gh-pages (hunk 1 above)
Comment (D4nte) [reply ci/Jenkinsfile.gh-pages:32]: Thank you but I don't understand why https://github.com/status-im/js-waku/pull/437 does not have any Jenkins build in the dashboard

**Comment by jakubgs** [comment]
This is not done, it's still failing to build: https://ci.status.im/job/website/job/js-waku.wakuconnect.dev/

**Comment by jakubgs** [comment]
Using `npm ci` creates moronic errors like these:
```
npm ERR! ENOTEMPTY: directory not empty, rename '/home/jenkins/workspace/website/js-waku.wakuconnect.dev/examples/eth-pm-wallet-encryption/node_modules/acorn-globals' -> '/home/jenkins/workspace/website/js-waku.wakuconnect.dev/examples/eth-pm-wallet-encryption/node_modules/.acorn-globals-h3wtysQn'
```
Which appear mostly random.

**Comment by jakubgs** [comment]
Using `npm install` causes the same issue. And it makes no sense because we call `cleanWs()` after every build:
```
 > a ci-slave-linux -a 'ls /home/jenkins/workspace/website/js-waku.wakuconnect.dev' 
linux-03.he-eu-hel1.slave.ci | FAILED | rc=2 >>
ls: cannot access '/home/jenkins/workspace/website/js-waku.wakuconnect.dev': No such file or directorynon-zero return code
linux-04.he-eu-hel1.slave.ci | FAILED | rc=2 >>
ls: cannot access '/home/jenkins/workspace/website/js-waku.wakuconnect.dev': No such file or directorynon-zero return code
linux-01.he-eu-hel1.slave.ci | FAILED | rc=2 >>
ls: cannot access '/home/jenkins/workspace/website/js-waku.wakuconnect.dev': No such file or directorynon-zero return code
linux-02.he-eu-hel1.slave.ci | FAILED | rc=2 >>
ls: cannot access '/home/jenkins/workspace/website/js-waku.wakuconnect.dev': No such file or directorynon-zero return code
```
So we always start with a fresh clone.

**Comment by D4nte** [comment]
Do you want me to convert to yarn see if it helps?

**Comment by jakubgs** [comment]
@D4nte should be done, check it out.

**Comment by D4nte** [comment]
@jakubgs awesome! I did not realize `file:` would make a difference (it did not locally)! well spotted.

**Pull request merged**